### ISSUE TITLE
Can't pull Docker Image 

### ISSUE THREAD

**Issue by riouruma** [docker/for-mac#5307]
- [x] I have tried with the latest version of Docker Desktop
  - [x] I have tried disabling enabled experimental features
  - [x] I have uploaded Diagnostics
  - Diagnostics ID:

### Information

  - macOS Version: Big Sur (M1)

#### Docker
```
$ docker pull php:7.4-fpm
Error response from daemon: Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on 192.168.64.1:53: read udp 192.168.64.3:51137->192.168.64.1:53: read: connection refused
```

#### DockerCompose

```
$ docker-compose build
mysql uses an image, skipping
nginx uses an image, skipping
Building php-fpm
Step 1/17 : FROM php:7.4-fpm
ERROR: Service 'php-fpm' failed to build : Get https://registry-1.docker.io/v2/: dial tcp: lookup registry-1.docker.io on 192.168.64.1:53: read udp 192.168.64.3:58037->192.168.64.1:53: read: connection refused
```

**Comment by tholu** [comment]
@riouruma Did you figure out the problem? I'm experiencing the same connectivity issues with the latest tech preview.

**Comment by hcancelik** [comment]
@tholu @riouruma are you guys using Laravel Valet? or installed dnsmasq via homebrew?

**Comment by tholu** [comment]
@hcancelik Yes, Laravel Valet was indeed the problem. See https://github.com/laravel/valet/issues/1031

**Comment by riouruma** [comment]
@tholu @hcancelik 
No, I'm not using Laravel Valet.
I'm still facing this issues.

I often write Ruby as well, and before time I had not got errors with Docker preview.
I have got same error when I try to build image on Ruby on Rails project as well.

**Comment by riouruma** [comment]
Thanks to @hcancelik, I could pull images.

1. stop `dnsmasq` (brew services stop dnsmasq)
2. restart Docker Desktop
3. $ docker pull php:7.4-fpm → will success

**Comment by kblcuk** [comment]
Fwiw in my case it was `Use new virtualization framework` experimental feature that caused this.

**Comment by djs55** [comment]
There were some fixes to the internal DNS in Docker Desktop 3.4.0 which I think should fix this issue. I'll close the ticket but if it recurs, upload a fresh set of diagnostics, quote the ID, ping me and I'll take a look / re-open the issue.
